overlays/hosts/internal/hosts_test.go

```diff
@@ -49,7 +49,7 @@ func Test_hostsOverlay(t *testing.T) {
 			assert.NoError(t, err)
 			assert.Empty(t, stdout.String())
 			assert.Empty(t, stderr.String())
-			assert.Equal(t, strings.Replace(tt.log, "%HOSTNAME%", hostname, -1), logbuf.String())
+			assert.Equal(t, tt.log, strings.Replace(logbuf.String(), hostname, "%HOSTNAME%", -1))
```

Comment:
You can either change the wanted to output so that it matches the output of the test. This is what was done in the original by you.
During the rebase, I found this disturbing and think its better to change the output of the test so that it matches the wanted output. Better cleaning than painting.

Reply:
Replacing the value of `hostname` in the log is less controlled. For example, if the output contained the string "We're off to a rocky start" and ran on an instance with hostname "rocky," it would erroneously change this to "we're off to a %HOSTNAME% start."